src/lib/input/input-container.ts

```diff
@@ -311,9 +295,8 @@ export class MdInputDirective implements OnChanges, OnDestroy, DoCheck {
   /** Re-evaluates the error state. This is only relevant with @angular/forms. */
   private _updateErrorState() {
     const oldState = this._isErrorState;
-    const control = this._ngControl;
-    const parent = this._parentFormGroup || this._parentForm;
-    const newState = control && this.errorStateMatcher(control.control as FormControl, parent);
+    const errorMatcher = this.errorStateMatcher || this._errorOptions.errorStateMatcher;
```

Comment:
Then instead of making the `Input()` take a function you make it take a `ErrorStateMatcher`. My other reason for not wanting `ErrorOptions` is that things like the stepper which just cares about the matcher then has to worry about all the other options too.